pkg/kubelet/stats/cri_stats_provider.go

```diff
@@ -206,7 +208,8 @@ func (p *criStatsProvider) makePodStats(podSandbox *runtimeapi.PodSandbox) *stat
 	}
 	podUID := types.UID(s.PodRef.UID)
 	if vstats, found := p.resourceAnalyzer.GetPodVolumeStats(podUID); found {
-		s.VolumeStats = vstats.Volumes
+		s.VolumeStats = append(vstats.EphemeralVolumes, vstats.PersistentVolumes...)
+		s.EphemeralStorage = calcEphemeralStorage(nil, vstats.EphemeralVolumes, rootFsInfo)
```

Comment:
Is there a reason we are not including container disk stats from the CRI here?

Reply:
this makePodStats is called inside of  ListPodStats() first for loop. In this for loop, it scans all the containers and find the pod for the given container and construct PodStats map (called sandboxIDToPodStats). If the pod is seen the first time, makePodStats is called and update the volumes information for the pod, and we calculates the ephemeral volume usage. But at this time,    in ListPodStats() loop, pod does not have all the containers information, we cannot pass containers here.
Two options here
1. Does not calculate the ephemeralstorage usage here. Wait until all containers are scanned from the for loop, then loop through the PodStats map, and calculates the storage, but we have to call resourceAnalyzer.GetPodVolumeStats again since podStats.Volumes containers all volumes in one list
2. Remove makePodStats call out of the first for loop in ListPodStats(). So we only need to call GetPodVolumeStats once. This would be similar to cadvisor_stats_provider.

Reply:
can we just pass the container stats as an argument to makePodStats?